### PR TITLE
MM-48640 Fix everywhere that mattermost-redux is importing web app code

### PR DESCRIPTION
**Pull request by hmhealey**
While mattermost-redux isn't fully set up as a proper package, we still need to ensure that it's not importing code from the web app because that causes import issues when plugins such as Playbooks import from it. Some of these changes are made to code that's quite old (around a year old) because we won't find them until we go to update those plugins to the latest version, so I also added an ESLint rule that should prevent this going forward.

Most of this can be solved by just moving code that needs access to the web app out of mattermost-redux itself or by moving constants into mattermost-redux. It's not a great system, but it's what we're stuck with until we can find time to spend most of a quarter blowing it all up and starting from scratch. See the individual commits for more information about each group of files that I changed.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48640

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: packages/mattermost-redux/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "rules": {
+    "no-restricted-imports": ["error", {
+      "patterns": [
+        {"group": ["actions/*", "selectors/*", "types/*", "utils/*", "!utils/test_helper"], "message": "mattermost-redux can't import files from the rest of the web app"}
```
Comment (lieut-data) [inline packages/mattermost-redux/.eslintrc.json:5]: ❤️

**Review by lieut-data — APPROVED** [review]
Wow, thank you @hmhealey -- way more effort than I was expecting on a Friday afternoon: I appreciate it :)

**Review by nevyangelova — APPROVED** [review]
Thanks for handling Harrison! 🎉

**Review by nevyangelova — APPROVED** [review]
Thanks for handling Harrison! 🎉

**Pull request merged**